pkg/storj/node.go

```diff
@@ -94,13 +94,15 @@ func (id NodeID) Less(b NodeID) bool {
 // Difficulty returns the number of trailing zero bits in a node ID
 func (id NodeID) Difficulty() (uint16, error) {
 	idLen := len(id)
-	for i := 1; i < idLen; i++ {
-		b := id[idLen-i]
+	var b byte
+	var zeroBits int
+	for i := 1; i <= idLen; i++ {
+		b = id[idLen-i]
 
 		if b != 0 {
-			zeroBits := bits.TrailingZeros16(uint16(b))
+			zeroBits = bits.TrailingZeros16(uint16(b))
 			if zeroBits == 16 {
-				zeroBits = 0
+				panic("impossible codepath")
```

Comment:
This function signature was modified to add an error return on recent past in favor of a panic. Unless you plan on changing the signature back, why not return an error instead?

Reply:
oh, good call!